projects/packages/autoloader/src/class-path-processor.php

```diff
@@ -121,7 +121,7 @@ private function get_normalized_constants() {
 	 * @return bool True if the path is absolute, otherwise false.
 	 */
 	private function is_absolute_path( $path ) {
-		if ( 0 === strlen( $path ) || '.' === $path[0] ) {
+		if ( empty( $path ) || 0 === strlen( $path ) || '.' === $path[0] ) {
```

Comment:
Is `0 === strlen( $path )` redundant here?

Reply:
The `empty()` itself is partially redundant too, since `$path` is always set.

I'd be more interested in what's calling this with a non-string in the first place. The one caller is passing a value returned from `wp_normalize_path()`, which should always be returning a string.

Reply:
> The one caller is passing a value returned from `wp_normalize_path()`, which should always be returning a string.

I was wondering about this as well. The only thing I could find is perhaps there's a scenario where `wp_normalize_path()` returns `null` due to a bad `preg_replace()` (e.g. UTF8 issues), but I wasn't able to reproduce right off.

Reply:
Even if that happened, the `return $wrapper . $path;` would wind up stringifying it in the end.